src/github.com/getlantern/peerscanner/web.go

```diff
@@ -51,36 +56,46 @@ func startHttp() {
 // register is the entry point for peers registering themselves with the service.
 // If peers are successfully vetted, they'll be added to the DNS round robin.
 func register(resp http.ResponseWriter, req *http.Request) {
-	name, ip, port, err := getHostInfo(req)
-	if err == nil && port != 443 {
-		err = fmt.Errorf("Port %d not supported, only port 443 is supported", port)
+	name, ip, port, supportedFronts, err := getHostInfo(req)
+	if err == nil && !(port == "80" || port == "443") {
+		err = fmt.Errorf("Port %d not supported, only ports 80 and 443 are supported", port)
```

Comment:
Peerscanner tests the servers by connecting to them directly by IP.  If we allow the hop between the CDN and the flashlight server to be unencrypted (and currently this is the only way I could get flashlight to work with Cloudfront so far), then the flashlight server will be listening in port 80.  The hop between the client and the CDN is of course always encrypted, over port 443, but peerscanner skips that.